relation_engine_server/utils/json_validation.py

```diff
@@ -22,7 +25,92 @@ def set_defaults(validator, properties, instance, schema):
                 instance.setdefault(property, subschema["default"])
         for error in validate_properties(validator, properties, instance, schema):
             yield error
+
     return validators.extend(validator_class, {"properties": set_defaults})
 
 
 Validator = extend_with_default(Draft7Validator)
+
+
+def get_schema_validator(schema=None, schema_file=None, validate_at=""):
+    """
+    Get a validator for the supplied schema
+
+    :param schema:      (dict)    the schema as a data structure
+    :param schema_file: (string)  path to the schema file (json or yaml format)
+
+    :param validate_at: (string)  where in the data structure the schema to validate against
+                                  is located, in JSON pointer syntax
+                                  defaults to the root of the schema object if not set
+
+    only one of `schema` and `schema_file` should be supplied
+
+    :return:
+    validator  (Validator)  jsonschema validator instance
+
+    """
+
+    if schema is None and schema_file is None or schema is not None and schema_file is not None:
```

Comment:
I'm not asking you to change this, but this could also be `if (schema is None) == (schema_file is None)`, which is a negated xor. Just thought that was kinda interesting :p